src/app/redux/productSlice.ts

```diff
@@ -0,0 +1,46 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+interface Product {
+  id: number;
+  name: string;
+  price: number;
+  image: string; 
+}
+
+interface ProductState {
+  products: Product[];
+}
+
+
+const image = {
+  apple: '/images/apple.jpg',
+  avocado: '/images/avacado.jpg',
+  burger: '/images/burger.jpg',
+  hotdog: '/images/hotdog.jpg',
+  melon: '/images/melon.jpg',
+  pizza: '/images/pizaa.jpg',
+  salad: '/images/salad.jpg',
+};
+
+const initialState: ProductState = {
+  products: [
```

Comment:
Why are the products listed in the redux slice? My assumption is those products are static/ immutable. They therefore should not be listed in the slice at all. Remove them from the productslice. You could add that object to the products page or better yrt put them in a json file and access them in the page where you need to access them